src/dma/mod.rs

```diff
@@ -608,79 +608,95 @@ macro_rules! dma_stream {
                 }
 
                 #[inline(always)]
-                fn clear_transfer_complete_interrupt(&mut self) {
+                fn clear_transfer_complete_flag(&mut self) {
                     //NOTE(unsafe) Atomic write with no side-effects and we only access the bits
                     // that belongs to the StreamX
                     let dma = unsafe { &*I::ptr() };
                     dma.$ifcr.write(|w| w.$tcif().set_bit());
                 }
 
                 #[inline(always)]
-                fn clear_half_transfer_interrupt(&mut self) {
+                fn clear_half_transfer_flag(&mut self) {
                     //NOTE(unsafe) Atomic write with no side-effects and we only access the bits
                     // that belongs to the StreamX
                     let dma = unsafe { &*I::ptr() };
                     dma.$ifcr.write(|w| w.$htif().set_bit());
                 }
 
                 #[inline(always)]
-                fn clear_transfer_error_interrupt(&mut self) {
+                fn clear_transfer_error_flag(&mut self) {
                     //NOTE(unsafe) Atomic write with no side-effects and we only access the bits
                     // that belongs to the StreamX
                     let dma = unsafe { &*I::ptr() };
                     dma.$ifcr.write(|w| w.$teif().set_bit());
                 }
 
                 #[inline(always)]
-                fn clear_direct_mode_error_interrupt(&mut self) {
+                fn clear_direct_mode_error_flag(&mut self) {
                     //NOTE(unsafe) Atomic write with no side-effects and we only access the bits
                     // that belongs to the StreamX
                     let dma = unsafe { &*I::ptr() };
                     dma.$ifcr.write(|w| w.$dmeif().set_bit());
                 }
 
                 #[inline(always)]
-                fn clear_fifo_error_interrupt(&mut self) {
+                fn clear_fifo_error_flag(&mut self) {
                     //NOTE(unsafe) Atomic write with no side-effects and we only access the bits
                     // that belongs to the StreamX
                     let dma = unsafe { &*I::ptr() };
                     dma.$ifcr.write(|w| w.$feif().set_bit());
                 }
 
                 #[inline(always)]
-                fn get_transfer_complete_flag() -> bool {
+                fn all_flags(&self) -> (bool, bool, bool, bool, bool)
```

Comment:
Maybe better return some structure. Tuple of booleans say nothing about what these flags mean.

Reply:
or use `bitflags`

Reply:
Agree, i will use a structure to keep things simple. Will do the same for other function that get or set several bits at once.

Have you an option about 'interupts' and 'set_interrupt' function ?

Anyway, my priority is to adapt code to make it compile. Stream and transfer are used in i2c module and the code here is also weird. I think this require to implement many Stream method on Transfer struct.

I also noticed that Stream and StreamISR trait may unnecessary. I mean, this is only implemented for <`StreamX<DMA, S> and implementation are already constrained to valid stream by `Sealed` ( but it 's better to use a `StreamInstance` marker trait to make the intend clear). However, I wil not change that.

Reply:
Yeees, the lib compile again